api/python/cellxgene_census/src/cellxgene_census/_release_directory.py

```diff
@@ -133,3 +160,31 @@ def get_census_version_directory() -> Dict[CensusVersionName, CensusVersionDescr
 
     # Cast is safe, as we have removed all aliases
     return cast(Dict[CensusVersionName, CensusVersionDescription], directory)
+
+
+def get_census_mirrors() -> CensusMirrors:
+    response = requests.get(CELL_CENSUS_MIRRORS_DIRECTORY_URL)
+    response.raise_for_status()
+    return cast(CensusMirrors, response.json())
+
+
+def _assert_mirror_supported(mirror: CensusMirror) -> None:
+    """
+    Verifies if the mirror is supported by this version of the census.
+    This method provides a proper error message in case an old version of the census
+    tries to connect to an unsupported mirror.
+    """
+    if mirror["provider"] not in SUPPORTED_PROVIDERS:
+        raise ValueError(f"Unsupported mirror provider: {mirror['provider']}. Try upgrading the census package.")
+
+
+def _resolve_census_locator(locator: CensusLocator, mirror: CensusMirror) -> ResolvedCensusLocator:
+    _assert_mirror_supported(mirror)
+
+    if locator.get("relative_uri"):
+        uri = _uri_join(mirror["base_uri"], locator["relative_uri"])
+        region = mirror["region"]
+    else:
+        uri = locator["uri"]
+        region = locator.get("s3_region")
```

Comment:
@ebezzi is it correct for this else branch to ignore `mirror`?

Reply:
Correct. The `else` branch can be removed as soon as we'll patch the `release.json` file with the new field though, so IMHO don't include that in the R API.